### PR TITLE
Add the option to specify the log event level on complete

### PR DESCRIPTION
**Pull request by IliyanAng**
There are use cases, where the complete operation needs to have different log levels depending on business logic. 
With this PR we are addressing this

### CODE REVIEW

File: src/SerilogTimings/Operation.cs [hunk 1]
```diff
@@ -181,6 +195,23 @@ public void Complete(string resultPropertyName, object result, bool destructureO
...
         }
 
+        /// <summary>
+        /// Complete the timed operation with an included result value and log event level.
+        /// </summary>
+        /// <param name="resultPropertyName">The name for the property to attach to the event.</param>
+        /// <param name="result">The result value.</param>
+        /// <param name="logEventLevel">The log event level with which the complete operation will be logged</param>
+        /// <param name="destructureObjects">If true, the property value will be destructured (serialized).</param>
+        public void Complete(string resultPropertyName, object result, LogEventLevel logEventLevel, bool destructureObjects = false)
```
Comment (nblumhardt) [inline src/SerilogTimings/Operation.cs:205]: The `result` and `destructureObjects` arguments are closely related, so perhaps the `level` argument could be first in this overload, instead?

Comment (nblumhardt) [reply src/SerilogTimings/Operation.cs:205]: Just `level` would be a suitable name for the `LogEventLevel` argument, following Serilog's argument naming.

**Review by nblumhardt — COMMENTED** [review]
Around line 280 we have:

```csharp
            level = elapsed > _warningThreshold?.TotalMilliseconds && level < LogEventLevel.Warning
                ? LogEventLevel.Warning
: level;
```

I think this will cause the `_warningThreshold` level to override the one passed explicitly, which seems slightly surprising.

What would you think about the explicit level overriding the default one in this case?

File: src/SerilogTimings/Operation.cs (hunk 1 above)
Comment (IliyanAng) [reply src/SerilogTimings/Operation.cs]: Sounds good, updated

Comment (IliyanAng) [reply src/SerilogTimings/Operation.cs]: Question: Isn't the override level only triggered when the requests are 'abandoned'? In which case you wouldn't have had explicitly given the level, as it never hit the Complete()

Comment (IliyanAng) [reply src/SerilogTimings/Operation.cs]: Actually, If you specify a warning threshold TimeSpan you would overwrite your Complete logging level. 

This is probably to be expected though, as lets say you want error logs, when it takes more than x amount of seconds, but if it's within expected time , fine tune the complete to whatever you want, like warning, or even error again, if the business logic demands it

Comment (nblumhardt) [reply src/SerilogTimings/Operation.cs]: Sorry about the lack of feedback/progress on this, I'm hoping to loop back around to it shortly (ended up a little snowed under the Serilog 3.1/.NET 8 changes).

**Pull request merged**